lib/engine/game/g_18_zoo/step/buy_sell_par_shares.rb

```diff
@@ -70,6 +70,25 @@ def available_companies
 
             @game.available_companies
           end
+
+          def round_state
+            # An extra field to track whether a share was already sold from any not president since the start of the round.
+            super.merge({ share_sold: [] })
+          end
```

Comment:
there is already a variable that keeps track of sold shares:
@round.players_sold

please use that (you can search for examples on how to use) before introducing new variables

Reply:
the variable is used to say if a player sold a specific company; it's used to be able to handle the "cannot buy if sold a company"
what I need to save here is the specific shares, because that specific Share should have a specific logic